### PR TITLE
NinjaBlocks API

### PR DESCRIPTION
**Pull request by jpgauthier**
- Implemented the api from the ground up
- Implemented everything except a few deprected/non-working
- Tested and working!
- Fully documented

### CODE REVIEW

File: app/apis/ninjablocks.js [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+var request = require('request'),
+	_ 		= require('lodash'),
+	util    = require('util'),
+    url     = 'https://api.ninja.is/rest/v0/';
+
+
+function ninjaBlocks(options) {
+
+	var self = this;
+  	this._options = options || {};
```
Comment (alexishuard) [inline app/apis/ninjablocks.js:16]: _.options is never used. Is this normal ?

File: app/apis/ninjablocks.js [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+	 * Returns the list of devices associated with the authenticating user.
+	 * @param  {Function} callback   Callback when request finished or error found
+	 */
+	this.devices = function(callback) {
+  		getRequest('devices', callback);
+  	};
+
+  	/**
+	 * Base definition for device.
+	 * @param  {String}   nodeId   Id of the device
```
Comment (alexishuard) [inline app/apis/ninjablocks.js]: Shouldn't it be guid instead of nodeId ?

**Comment by alexishuard** [comment]
Very well done @jpgauthier ! Really well wrapped and documented. This API will be a charm to use. Thanks !

:+1:

**Comment by simonbenoit** [comment]
Nice work :shipit:

**Comment by krapskill** [comment]
:+1:

**Pull request merged**